SazhnevDjango/SazhnevProject/sazhnevapp/views.py

```diff
@@ -11,30 +14,68 @@ def wrapper(*args, **kwargs):
+             "Количество": el.quantity,
+             "Цена": el.price,
+             "Дата завоза": el.added_date}
+        )
+    return pd.DataFrame(table).to_html(index=False)
+
+
 @log
 def main(request):
-    html_main = ("<html>\n<head>\n<title>Главная</title>\n</head>"
-                 "\n<body>\n<h1>Домашняя страница!</h1>\n<div>\n"
-                 "Добро пожаловать на нашу главную страницу! "
-                 "Здесь вы найдете все необходимые информацию о нашей компании, наших услугах и продукции. "
-                 "Мы стремимся предоставить вам лучший сервис и качественные товары, "
-                 "чтобы удовлетворить все ваши потребности. "
-                 "Наша команда профессионалов всегда готова помочь вам и ответить на все ваши вопросы."
-                 "Спасибо, что выбрали нас!"
-                 "\n</div>\n</body>\n</html>\n")
-    return HttpResponse(html_main)
+    return render(request, "base.html")
 
 
 @log
 def about_me(request):
-    my_page = ("<html>\n<head>\n<title>О себе</title>\n</head>"
-               "\n<body>\n<h1>Моя страница!</h1>\n<div>\n"
-               "<i>Добро пожаловать на страницу обо мне! "
-               "Я Сажнев Алексей и мне 33 года. "
-               "Живу в Краснодарском крае и являюсь студентом GeekBrains."
-               "Целеустремлён,общителен и творчески развит.</i>"
+    my_page = ("<html>\n<head>\n<title>О нас</title>\n</head>"
+               "\n<body>\n<h1>О нас!</h1>\n<div>\n"
+               "<i>Мы - команда людей, которые любят заботиться о своем здоровье и здоровье своих клиентов. "
+               "Наш интернет магазин 'Овощи и фрукты' был создан с целью "
+               "предоставить качественные и свежие продукты прямо к вашему столу.</p>"
+               "<p>Мы работаем напрямую с производителями, чтобы обеспечить наших клиентов только лучшими овощами и фруктами."
+               "Мы следим за качеством продукции и стремимся предложить широкий ассортимент товаров для разнообразия вашего стола.</p>"
+               "<p>Наша цель - сделать вашу жизнь здоровее и легче, предлагая качественные продукты, удобный сервис доставки и отличное обслуживание.</p>"
+               "<p>Присоединяйтесь к нашему магазину 'Овощи и фрукты'"
+               "и наслаждайтесь свежими и вкусными продуктами каждый день!</p></i>"
                "\n</div>\n</body>\n</html>\n")
     return HttpResponse(my_page)
+
+
+def get_catalog(request):
+    product = Product.objects.all()
+    result = get_html_table(product)
+    return render(request, 'sazhnevapp/catalog.html', {'table': result})
+
+
+from datetime import datetime, timedelta
+from django.db.models import Count
+
+
+def get_orders(request, client_id):
+    client = get_object_or_404(Client, pk=client_id)
+    """Заказы за неделю"""
+    weekly_orders = Order.objects.filter(client=client,
+                                         order_date__gte=datetime.now() - timedelta(days=7)).order_by('order_date')
```

Comment:
Можно извлекать напрямую из Product, используя обратную связь
```suggestion
    weekly_orders = Product.objects.filter(order__client=client, order__order_date__gte=datetime.now() - timedelta(days=7)).distinct()
```
P.S. Возможно сделка ошибся в синтаксисе, но логика верна